day_01/src/main.rs

```diff
@@ -24,12 +24,46 @@ fn task_1(reader: &mut BufReader<File>) -> u32 {
     sum
 }
 
+fn task_2(reader: &mut BufReader<File>) -> u32 {
+    let mut sum = 0;
+
+    for line in reader.lines() {
+        let line = line.unwrap();
+
+        // stole this part from christ biscardi
+        let found_numbers = &line
+            .replace("one", "one1one")
+            .replace("two", "two2two")
+            .replace("three", "three3three")
+            .replace("four", "four4four")
+            .replace("five", "five5five")
+            .replace("six", "six6six")
+            .replace("seven", "seven7seven")
+            .replace("eight", "eight8eight")
+            .replace("nine", "nine9nine")
+            .chars()
+            .filter(|c| c.is_digit(10))
+            .collect::<Vec<char>>();
+        // println!(" {:?}", &found_numbers);
+        let combined_number = format!(
+            "{}{}",
+            found_numbers.first().unwrap().to_string(),
+            found_numbers.last().unwrap().to_string()
+        )
+        .parse::<u32>()
+        .unwrap();
+
+        sum += combined_number;
+    }
+    sum
+}
+
 fn main() {
     let mut example_input = BufReader::new(File::open("./example_1.txt").unwrap());
     let mut full_input = BufReader::new(File::open("./puzzle_input.txt").unwrap());
 
-    println!("example data: total sum is {}", task_1(&mut example_input));
+    // println!("example data: total sum is {}", task_1(&mut example_input));
```

Comment:
Please remove the commented code. It's in the history anyway if you need it.

Reply:
as always - you're right ...